src/oas/generator.gleam

```diff
@@ -1071,32 +1073,57 @@ fn defs(xs) {
 
 pub fn build(spec_src, project_path, provider, exclude) {
   let module_path = project_path <> "/src/" <> provider
-  use spec <- try(fs.read_json(spec_src, oas.decoder))
+  use file <- try(
+    simplifile.read(spec_src)
+    |> snag.map_error(fn(_) { "Unable to read " <> spec_src }),
+  )
+
+  use spec <- try(
+    json.decode(file, oas.decoder)
+    |> snag.map_error(json_decode_error_to_string),
+  )
 
   let #(operations, entry) =
     gen_operations_and_top_files(spec, provider, exclude)
 
-  use Nil <- try(fs.write(
-    module_path <> "/operations.gleam",
-    operations |> bit_array.from_string,
-  ))
+  let operations_module_file = module_path <> "/operations.gleam"
+
+  use Nil <- try(
+    simplifile.write_bits(
+      operations_module_file,
+      operations |> bit_array.from_string,
+    )
+    |> snag.map_error(fn(_) { "Unable to write to " <> operations_module_file }),
+  )
+
+  use mod <- try(
+    simplifile.read(module_path <> ".gleam")
+    |> snag.map_error(fn(_) { "Unable to read " <> module_path <> ".gleam" }),
```

Comment:
Could we not throw away the detail of the error message. It would be helpful to have.
I had used this style with snag in midas_node. 
The actual filesystem error was the core error and the file being read the context
https://github.com/midas-framework/midas_node/blob/main/src/midas/node/file_system.gleam#L59-L60